subworkflows/local/quilt_subworkflows/impute_quilt.nf

```diff
@@ -0,0 +1,54 @@
+include { QUILT_QUILT     } from '../../../modules/nf-core/quilt/quilt/main'
+
+workflow IMPUTE_QUILT {
+
+    take:
+    ch_hap_legend                            // channel: [ val(meta),hap, legend ]
+    ch_input                                //  channel: [ val(meta),bam, bai ]
+    ch_chunks                              //   channel: [ val(meta), start_coordinate, end_coordinate, number ]
+
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    posfile = []
+    phasefile = []
+    posfile_phasefile = [[id: null], posfile, phasefile]
+    genetic_map_file = []
+    fasta = [[id:'test'], []]
+
+    ngen = params.ngen
+    buffer = params.buffer
+
+    ch_bam_bamlist = ch_input
+
+    if (genetic_map_file.isEmpty()) {
+        ch_hap_chunks = ch_hap_legend.combine(ch_chunks, by:0).map { it + ngen + buffer + [[]] }
+    } else {
+        println "genetic_map_file is not empty"
+        // Add ngen and buffer
+        ch_hap_chunks = ch_hap_legend.join(ch_chunks, by:0).join(genetic_map_file)
```

Comment:
The by 0 is the default option and could be removed.

Reply:
As you initialise the genetic_map_file in the process, wouldn't it always be null ?

Reply:
Indeed, that's the case. This case (providing a genetic map) is not tested and will be developed in the future.